### PR TITLE
PXC-2991: Block group replication when pxc_strict_mode is enforced

### PR DESCRIPTION
**Pull request by dutow**

### CODE REVIEW

**Comment by it-percona** [comment]
[CLA assistant check](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster?pullRequest=959) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster?pullRequest=959) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/percona-xtradb-cluster?pullRequest=959) it.</sub>

File: sql/wsrep_var.cc [hunk 1]
```diff
@@ -930,12 +932,14 @@ bool pxc_strict_mode_check(sys_var *, THD *thd, set_var *var) {
...
 
+    const bool gr_running = is_group_replication_running();
+
     /* replicate_myisam = off
        row_binlog_format = true (row)
        safe_log_output = true (none/file)
        serializable = false */
     block = !(!replicate_myisam && row_binlog_format && safe_log_output &&
-              !serializable);
+              !serializable && !gr_running);
```
Comment (kennt-percona) [inline sql/wsrep_var.cc:943]: should update the comment above also

File: sql/wsrep_var.cc [hunk 2]
```diff
@@ -953,14 +957,20 @@ bool pxc_strict_mode_check(sys_var *, THD *thd, set_var *var) {
           "Can't change pxc_strict_mode while isolation level is"
           " SERIALIZABLE");
 
+    if (gr_running)
+      WSREP_ERROR(
+          "Can't change pxc_strict_mode with group replication "
```
Comment (kennt-percona) [inline sql/wsrep_var.cc]: should be "when group replication is running"

File: sql/rpl_group_replication.cc [hunk 3]
```diff
@@ -95,7 +97,19 @@ LEX_CSTRING group_replication_plugin_name_str = {
...
-int group_replication_init() { return initialize_channel_service_interface(); }
+int group_replication_init() {
+#ifdef WITH_WSREP
+  if (pxc_strict_mode >= PXC_STRICT_MODE_ENFORCING) {
+    WSREP_ERROR(
+        "Group replication cannot be used with Percona XtraDB Cluster in "
+        "strict mode.");
+    return 1;
+  } else {
+    WSREP_WARN("Using group replication with Percona XtraDB Cluster.");
```
Comment (kennt-percona) [inline sql/rpl_group_replication.cc]: Needs more explanation here, or a more active sounding warning.

**Review by kennt-percona — APPROVED** [review]
lgtm

**Comment by dutow** [comment]
https://pxc.cd.percona.com/view/PXC%208.0/job/pxc-8.0-param/9/

**Pull request merged**